server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1578,7 +1585,15 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
         } else if ("remove".equalsIgnoreCase(operation)) {
             List<Long> accountIds = new ArrayList<Long>();
             for (String accountName : accountNames) {
-                Account permittedAccount = _accountDao.findActiveAccount(accountName, domain.getId());
+                Account permittedAccount = null;
+                if (isCallerRootAdmin) {
+                    List<AccountVO> accountsWthName = _accountDao.findAccountsLike(accountName);
+                    if (accountsWthName.size() > 0) {
+                        permittedAccount = accountsWthName.get(0);
```

Comment:
@anuragaw could this cause regression or issues, this is always picking up the first account in the list; should it be something like - if user is root admin, then all enable/permit for all accounts matching the name using something like `accountIds.add(permittedAccount.getId());` in the if block; and refactor suitably?

Reply:
No longer part of PR as above.